Lib/urllib/parse.py

```diff
@@ -947,9 +947,9 @@ def splithost(url):
     """splithost('//host[:port]/path') --> 'host[:port]', '/path'."""
     global _hostprog
     if _hostprog is None:
-        _hostprog = re.compile('//([^/#?]*)(.*)', re.DOTALL)
+        _hostprog = re.compile('//([^/#?\n]*)(.*)')
```

Comment:
This will caused just returning a tuple `(None, url)` as in the case when the host is not specified. The second item can contain `'\n'`. This doesn't mean `splithost()` rejects the url.

If you want to make `splithost()` rejecting URLs with newlines, check explicitly `'\n' in url` and raise an exception. But I think this is not the best place of doing such checks.